### DISCUSSION TITLE
Dynamic map creation/download

### DISCUSSION THREAD

**Discussion by abauske** [mapsforge/mapsforge#1202] [Q&A] [answered]
I have just stumbled upon this project and it looks exactly what I am looking for!
There is one question I could not find an answer for in the docs:
Is it possible to have the map downloaded dynamically depending on your current view position and zoom level? All I've seen seems to me like a fixed map that is displayed. My application for mapsforge however would require a huge region to be shipped with the application. However each user probably would only use a tiny part of the map data.

Could you please nudge me in the right direction whether I would have to implement such feature on my own or if it is included somewhere?

Thanks in advance!

**Comment by devemux86** [comment]
You probably mean a server providing vector tiles.
Check [applications list](https://github.com/mapsforge/mapsforge/blob/master/docs/Mapsforge-Applications.md), there are some tile server examples.

**Comment by devemux86** [comment, marked as answer]
However Mapsforge was designed for offline vector map files.

To create online vector tiles, there are more suitable tools if you search the internet.

**Reply by abauske** [reply]
Thanks for your quick response!
This is probably not the correct place to ask, but while we're at it: Any chance you have an suggestion on a tool (plain "non Android" Java)? I tried a couple of them however I haven't found anything suitable yet (some don't support touch screen and many more problems), so I would be glad if anyone could give a hint for a good tool.

**Comment by devemux86** [comment]
You may check [VTM](https://github.com/mapsforge/vtm) an OpenGL map library (compatible with Mapsforge).

It runs on Android + Desktop and supports online vector tiles.
There are examples per platform in its repository.
